frontend/src/components/nodes/Nodes.tsx

```diff
@@ -1,61 +1,101 @@
   };
 
-  componentDidUpdate(prevProps: any) {
-    if (this.props.role !== prevProps.role) {
-      this.autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
+  signNewValidators = (newValidators: any) => {
+    for (let i = 0; i < newValidators.length; i++) {
+      newValidators[i].new = true;
     }
-  }
+    return newValidators;
+  };
 
-  config = {
-    fetchDataFn: this.fetchNodes,
-    count: this.props.count,
-    category: "Node",
+  signRemovedValidators = (removedValidators: any) => {
+    for (let i = 0; i < removedValidators.length; i++) {
+      removedValidators[i].removed = true;
+    }
+    return removedValidators;
   };
 
-  autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
+  getNodes = async () => {
+    if (this.props.role === "validators") {
+      let nodes = await new NodesApi().queryNodeRpc();
+      let currentValidators = nodes.current_validators;
+      let nextValidators = nodes.next_validators;
+      let diffEpochValidators = VA.diffEpochValidators(
+        currentValidators,
+        nextValidators
+      );
+      console.log(diffEpochValidators);
+      let newValidators = this.signNewValidators(
+        diffEpochValidators.newValidators
+      );
+      this.signRemovedValidators(diffEpochValidators.removedValidators);
+      currentValidators = currentValidators.concat(newValidators);
+      this.setState({ validators: currentValidators });
+    }
+    if (this.props.role === "online-nodes") {
+      let onlineNodes = await this.getNonValidatingNodes();
+      this.setState({ onlineNodes });
+    }
+    if (this.props.role === "proposals") {
+      let proposals = (await new NodesApi().queryNodeRpc()).current_proposals;
+      this.setState({ proposals });
+    }
+  };
 
-  render() {
-    return <this.autoRefreshNodes />;
+  componentDidUpdate(prevProps: any) {
+    if (this.props.role !== prevProps.role) {
+      this.getNodes();
+    }
   }
-}
 
-interface InnerProps {
-  items: N.NodeInfo[];
-}
+  componentDidMount() {
+    this.getNodes();
+  }
 
-class Nodes extends React.Component<InnerProps> {
   render() {
-    const { items } = this.props;
-    return (
-      <FlipMove duration={1000} staggerDurationBy={0}>
-        {items &&
-          items.map((node) => <NodeRow key={node.nodeId} node={node} />)}
-      </FlipMove>
-    );
+    const { validators, onlineNodes, proposals } = this.state;
+    let nodes;
+    if (validators && this.props.role === "validators") {
+      nodes = validators.map((node: N.Validating) => (
+        <ValidatorRow key={node.account_id} node={node} />
+      ));
+    }
+    if (onlineNodes && this.props.role === "online-nodes") {
+      nodes = onlineNodes.map((node: N.NodeInfo) => (
+        <NodeRow key={node.nodeId} node={node} />
+      ));
+    }
+    if (proposals && this.props.role === "proposals") {
+      nodes = proposals.map((node: N.Proposal) => (
+        <ProposalRow key={node.account_id} node={node} />
+      ));
```

Comment:
BTW, you have re-invented routing in this component :man_shrugging: 

It might make more sense to have different components and even different pages (I mean different files in `pages/nodes/` folder instead of `[role].jsx`)

Reply:
will make them different page